router-ipv6 cmake compile error

if compile with cmake and BACDL_BIP=ON and BACDL_BIP6=ON

```
duplicate symbol '_Device_Object_Instance_Number'
```

I think the function can be removed, but then Makefile will no longer work.

The Device_Object_Instance_Number come from libbacnet-stack.a
With Linux and Apple.
```
[build] [ 20%] Building C object CMakeFiles/bacnet-stack.dir/src/bacnet/basic/object/device.c.o
...
[build] [ 86%] Building C object CMakeFiles/router-ipv6.dir/apps/router-ipv6/main.c.o
...
[build] [ 91%] Linking C executable router-ipv6
...
[build] duplicate symbol '_Device_Object_Instance_Number' in:
[build]     /Users/patrick/bacnet-stack-upstream-1/build/CMakeFiles/router-ipv6.dir/apps/router-ipv6/main.c.o
[build]     libbacnet-stack.a[57](device.c.o)
[build] ld: 1 duplicate symbols
[build] clang: error: linker command failed with exit code 1 (use -v to see invocation)
[build] gmake[2]: *** [CMakeFiles/router-ipv6.dir/build.make:98: router-ipv6] Error 1
[build] gmake[1]: *** [CMakeFiles/Makefile2:785: CMakeFiles/router-ipv6.dir/all] Error 2
[build] gmake[1]: *** Waiting for unfinished jobs....
```
build.make
```
# External object files for target router-ipv6
router__ipv6_EXTERNAL_OBJECTS =

router-ipv6: CMakeFiles/router-ipv6.dir/apps/router-ipv6/main.c.o
router-ipv6: CMakeFiles/router-ipv6.dir/build.make
router-ipv6: libbacnet-stack.a
router-ipv6: CMakeFiles/router-ipv6.dir/link.txt
	@$(CMAKE_COMMAND) -E cmake_echo_color "--switch=$(COLOR)" --green --bold --progress-dir=/Users/patrick/bacnet-stack-upstream-1/build/CMakeFiles --progress-num=$(CMAKE_PROGRESS_2) "Linking C executable router-ipv6"
	$(CMAKE_COMMAND) -E cmake_link_script CMakeFiles/router-ipv6.dir/link.txt --verbose=$(VERBOSE)
```
link.txt
```
/usr/bin/clang -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names "CMakeFiles/router-ipv6.dir/apps/router-ipv6/main.c.o" -o router-ipv6  libbacnet-stack.a
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -945,11 +945,8 @@ if(BACNET_STACK_BUILD_APPS)
       router-ipv6
       PRIVATE ${PROJECT_NAME})
     target_compile_options(router-ipv6 PRIVATE
-      # These make this example not totally C90 compatible but it is ok.
-
-      -Wno-declaration-after-statement
-      -Wno-overlength-strings
-      -Wno-variadic-macros
+      # Unreachable code because we have endless loop.
+      $<$<C_COMPILER_ID:MSVC>:/wd4702>
     )
   endif()
 

--- a/apps/router-ipv6/Makefile
+++ b/apps/router-ipv6/Makefile
@@ -13,7 +13,9 @@ TARGET_BIN = ${TARGET}$(TARGET_EXT)
 # BACNET_SRC_DIR is defined in common apps Makefile
 BACNET_OBJECT_DIR = $(BACNET_SRC_DIR)/bacnet/basic/object
 SRC = main.c \
-	$(BACNET_SRC_DIR)/bacnet/basic/npdu/s_router.c
+	$(BACNET_SRC_DIR)/bacnet/basic/npdu/s_router.c \
+	$(BACNET_OBJECT_DIR)/client/device-client.c \
+	$(BACNET_OBJECT_DIR)/netport.c
 
 # WARNINGS, DEBUGGING, OPTIMIZATION are defined in common apps Makefile
 # BACNET_DEFINES is defined in common apps Makefile

--- a/apps/router-ipv6/main.c
+++ b/apps/router-ipv6/main.c
@@ -26,6 +26,7 @@
 #include "bacnet/basic/sys/debug.h"
 #include "bacnet/basic/tsm/tsm.h"
 #include "bacnet/basic/binding/address.h"
+#include "bacnet/basic/object/device.h"
 #include "bacnet/basic/services.h"
 /* port agnostic file */
 #include "bacport.h"
@@ -39,7 +40,6 @@
 
 /* current version of the BACnet stack */
 static const char *BACnet_Version = BACNET_VERSION_TEXT;
-static uint32_t Device_Instance_Number = BACNET_MAX_INSTANCE;
 
 /**
  * 6.6.1 Routing Tables
@@ -1130,17 +1130,6 @@ static void control_c_hooks(void)
 #endif
 
 /**
- * @brief Get the Device object instance number
- * @return The Device object instance number
- * @note This is a proxy function to satisfy the BACnet Stack IPv6 port layer
- * requirements since this application omits a Device object.
- */
-uint32_t Device_Object_Instance_Number(void)
-{
-    return Device_Instance_Number;
-}
-
-/**
  * Main function of simple router demo.
  *
  * @param argc [in] Arg count.
@@ -1158,7 +1147,7 @@ int main(int argc, char *argv[])
     printf("BACnet Simple IP to IPv6 Router Demo\n");
     printf("BACnet Stack Version %s\n", BACnet_Version);
     if (argc > 1) {
-        Device_Instance_Number = strtol(argv[1], NULL, 0);
+        Device_Set_Object_Instance_Number(strtol(argv[1], NULL, 0));
     }
     datalink_init();
     atexit(cleanup);

